Src/Extern/HDF5/AMReX_ParticleHDF5.H

```diff
@@ -1299,9 +1307,19 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
     host_int_attribs.resize(finest_level_in_file+1);
 
     for (hsize_t i = 0; i < cnt; i++) {
-        p.id()   = iptr[0];
-        p.cpu()  = iptr[1];
-
+        if (convert_ids) {
+            union punner {
+                std::int32_t i;
+                std::uint32_t u;
+            };
+            punner x, y;
+            x.i = iptr[0];
+            y.i = iptr[1];
+            p.m_idcpu = ((std::uint64_t)x.u) << 32 | y.u;
```

Comment:
I think this is one of those things that this in principal UB but that in practice compilers basically have to make work or a ton of real-world code would break. I agree it's better to avoid if possible - can you suggest a way of implementing this that avoids UB?